### PR TITLE
`cf logs` and `cf logs --recent` now pull logs from log-cache

### PR DESCRIPTION
**Pull request by heycait**
## Does this PR modify CLI v6 or v7?

This PR modifies CLI v6.

## Description of the Change
Convert `cf logs` command to read and stream logs from log-cache instead of traffic controller. The output of the logs/UI remains the same. 

## Why Is This PR Valuable?

`cf logs` used to get its logs from Traffic Controller which is deprecated and will be removed in the fall. Once it's removed, `cf logs` would no longer work without these changes.

## Why Should This Be In Core?

This functionality already existed in cf CLI, we are only replacing the source of where the logs come from.

## How Urgent Is The Change?
Moderate. 

cc/ @cloudfoundry/cf-loggregator

### CODE REVIEW

**Comment by staylor14** [comment]
Hey @heycait && @kitemongerer !

We completed [an exploration](https://www.pivotaltracker.com/story/show/166679710) of how merging this change would affect the cli.  Applying this patch introduces an existing problem in the old logcache plugin to all of the commands that this PR affects.

An illustration of the problem and the steps to reproduce are included in the story.

Can you take a look and let us know what the next steps might be to resolve the problem?  Please feel free to hit us up on Slack for a more immediate response.

Regards,
Steve

**Comment by kitemongerer** [comment]
@staylor14 Could we have a little more detail on what the bug is and how to reproduce it? Have you done any validation that the current implementation of `cf logs --recent` isn't impacted by this bug?

**Comment by bwasmith** [comment]
Closing this PR for now as the work has been put down until a later date. Please feel free to reopen when we are ready to continue discussions on how to switch over the CLI to use LogCache

File: command/v7/push_command.go [hunk 1]
```diff
@@ -26,6 +26,14 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// NOAAClient is a client for getting logs.
+type NOAAClient interface {
```
[the remark this answers is not in the document]
Comment (MasslessParticle) [reply command/v7/push_command.go]: Our goal at the time was only the v6 CLI. We refactored the shared stuff into the loggingaction package. Afterward, only the v7 stuff referred to trafficcontroller.

**Comment by heyjcollins** [comment]
Closed as we've implemented this internally.

**Pull request closed without merging**